cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java

```diff
@@ -87,82 +80,38 @@ public void initialize(AuthorizationContext context) throws Exception {
   }
 
   @Override
-  public void grant(EntityId entityId, Principal principal, Set<Action> actions) throws Exception {
-    LOG.trace("Granting {} on {} to {}", actions, entityId, principal);
-    switch (principal.getType()) {
-      case ROLE:
-        binding.grant(entityId, new Role(principal.getName()), actions, getRequestingUser());
-        break;
-      case USER:
-        // get the group for the user to perform group based grant
-        performGroupBasedGrant(entityId, getGroupPrincipal(principal), actions);
-        break;
-      case GROUP:
-        performGroupBasedGrant(entityId, principal, actions);
-        break;
-      default:
-        throw new IllegalArgumentException(
-          String.format("The given principal '%s' is of unsupported type '%s'.", principal.getName(),
-                        principal.getType()));
+  public void grant(co.cask.cdap.proto.security.Authorizable authorizable, Principal principal, Set<Action> actions)
+    throws Exception {
+    LOG.trace("Granting {} on {} to {}", actions, authorizable, principal);
+    if (principal.getType().equals(Principal.PrincipalType.ROLE)) {
+      binding.grant(authorizable, new Role(principal.getName()), actions, getRequestingUser());
+      LOG.trace("Granted {} on {} to {}", actions, authorizable, principal);
+    } else {
+      throw new IllegalArgumentException(
+        String.format("The given principal '%s' is of unsupported type '%s'.", principal.getName(),
```

Comment:
Let's say - `Sentry only supports granting privileges to a role. The given principal...`